src/Api/PubnubApi/EventEngine/Subscribe/SubscribeEventEngine.cs

```diff
@@ -22,8 +28,13 @@ public class SubscribeEventEngine : Engine {
 			dispatcher.Register<Invocations.ReceiveReconnectInvocation, Effects.ReceivingEffectHandler>(receiveHandler);
 			dispatcher.Register<Invocations.CancelReceiveMessagesInvocation, Effects.ReceivingEffectHandler>(receiveHandler);
 
-			// TODO: ReconnectionConfiguration
-			currentState = new UnsubscribedState();
+			var emitMessageHandler = new Effects.EmitMessagesHandler(pubnubInstance, messageListener);
+			dispatcher.Register<Invocations.EmitMessagesInvocation, Effects.EmitMessagesHandler>(emitMessageHandler);
+
+			var emitStatusHandler = new Effects.EmitStatusEffectHandler(pubnubInstance, statusListener);
```

Comment:
Do we need link `statusListener` and `messageListener` with `pubnubInstance`  ? Are we not instantiating `SubscribeEventEngine` per Pubnub instance.

Reply:
Pubnub reference was needed as all listener delegates signature are : (Pubnub instance, ...).  So while calling those delegates along with data effect handler got, It has to pass pubnub instance reference along with it.
Other then that no other relation handler has with pubnub instance.

Reply:
I'm referring this [line](https://github.com/pubnub/c-sharp/blob/ee/integration/src/Api/PubnubApi/EventEngine/Subscribe/Effects/EmitMessagesHandler.cs#L36)